tests/Fixer/ClassNotation/VisibilityRequiredFixerTest.php

```diff
@@ -595,4 +597,91 @@ function A()
             '
         );
     }
+
+    public function testRemovingNewlinesBetweenKeywords()
+    {
+        $this->doTest(
+            '<?php
+                class Foo
+                {
+                    public $bar;
+
+                    final public static function bar() {}
+
+                    final public static function baz() {}
+                }',
+            '<?php
+                class Foo
+                {
+                    var
+                    $bar;
+
+                    final
+                    public
+                    static
+                    function bar() {}
+
+                    static
+                    final
+                    function baz() {}
+                }'
+        );
+    }
+
+    /**
+     * @requires PHP 7.1
+     */
+    public function testKeepingComment()
+    {
+        $this->fixer->configure(['elements' => ['property', 'method', 'const']]);
+
+        $this->doTest(
+            '<?php
+                class Foo
+                {
+                    /* constant */ private const BAR = 3;
```

Comment:
Why swapping the comment with `private`?

Reply:
Because of the checking in function `isKeywordPlacedProperly`.

Do you think I should change it to:
```php
    private function isKeywordPlacedProperly(Tokens $tokens, $keywordIndex, $comparedIndex)
    {
        for ($index = $comparedIndex - 1; $index > $keywordIndex; --$index) {
            if (!$tokens[$index]->isGivenKind([T_COMMENT, T_DOC_COMMENT, T_WHITESPACE])) {
                return false;
            }
            if (1 === Preg::match('/\R/', $tokens[$index]->getContent())) {
                return false;
            }
        }

        return true;
    }
```
?

Reply:
I don't know the implementation so I cannot answer this question properly for now. All I'm saying is that having such change when running the tool on a codebase would be quite unexpected.

Reply:
You can see `isKeywordPlacedProperly` function in this PR - I've extracted the logic to it for easier understanding.

> having such change when running the tool on a codebase would be quite unexpected.

Would it be? Current behaviour is removing such comment anyway.

Reply:
it's better to move the comment rather than dropping it, for sure :+1: 
it's not perfect indeed, I agree, yet input code is not perfect as well, and we are making it slightly better